app/tests/test_search_utils.py

```diff
@@ -0,0 +1,167 @@
+def test_execute_search(mock_open_search, app):
+    with app.app_context():
+        dsl_query = {"query": {"match_all": {}}}
+        execute_search(mock_open_search, dsl_query, page=1, per_page=10)
+        mock_open_search.search.assert_called_once()
+
+
+@patch("app.main.util.pagination.calculate_total_pages", return_value=10)
+@patch("app.main.util.pagination.get_pagination", return_value={"page": 1})
+def test_get_pagination_info(mock_calculate_total_pages, mock_get_pagination):
+    results = {"hits": {"total": {"value": 100}}}
+    total_records, pagination = get_pagination_info(results, 1, 10)
+    assert total_records == 100
+    assert pagination == {
+        "next": 2,
+        "pages": [1, 2, "ellipses", 10],
+        "previous": None,
+    }
+
+
+@pytest.mark.parametrize(
+    "to_filter, filter, expected",
+    [
+        (["a", "b", "c"], ["b"], ["a", "c"]),
+        (["a", "b", "c"], [], ["a", "b", "c"]),
+        ([], ["a", "b"], []),
+        ([], [], []),
+        (["a", "b", "c"], ["a", "b", "c"], []),
+        (["a", "b", "c", "d"], ["b", "d"], ["a", "c"]),
+        (["a", "b", "c"], ["x", "y", "z"], ["a", "b", "c"]),
+        (["a", 1, "b", 2, "c"], ["a", 2], [1, "b", "c"]),
+        ([1, 2, 3, "a", "b"], [2, "a"], [1, 3, "b"]),
+    ],
+)
+def test_get_filtered_list(to_filter, filter, expected):
+    assert get_filtered_list(to_filter, filter) == expected
+
+
+@patch("app.main.util.search_utils.OpenSearch")
+def test_get_all_fields_excluding(mock_open_search):
+    mock_open_search.indices.get_mapping.return_value = {
+        "documents": {
+            "mappings": {
+                "properties": {
+                    "field1": {},
+                    "field2": {},
+                    "field3": {},
+                }
+            }
+        }
+    }
+    fields = get_all_fields_excluding(mock_open_search, "documents", ["field2"])
+    mock_open_search.indices.get_mapping.assert_called_once()
+    assert fields == ["field1", "field3"]
+
+
+@patch(
+    "app.main.util.search_utils.get_all_fields_excluding",
+    return_value=["fieldA", "fieldB"],
+)
+def test_get_query_multi_match(mock_get_all_fields_excluding):
+    query = "test_query"
+    search_area = "metadata"
+    result = get_query_multi_match(
+        query, search_area, MockOpenSearch(), _id="12345"
+    )
+    assert result["query"]["bool"]["must"][0]["multi_match"]["query"] == query
+    assert (
+        "fieldA" in result["query"]["bool"]["must"][0]["multi_match"]["fields"]
+    )
+    assert (
+        result["query"]["bool"]["filter"][0]["term"][
+            "transferring_body_id.keyword"
+        ]
+        == "12345"
+    )
+
+
+def test_get_query_source_rules():
+    source_rules = get_query_source_rules()
+    assert source_rules == {"_source": {"exclude": ["*.keyword"]}}
+
+
+@patch(
+    "app.main.util.search_utils.get_query_source_rules",
+    return_value={"_source": {"exclude": ["*.keyword"]}},
```

Comment:
seems unnecessary to mock a lot of these functions which are just hardcoded values., e.g. `get_query_source_rules` `get_query_highlighting`,`get_query_aggregations` etc.

Tbh you could even do away with the functions as they only get used here and are pretty simple/ could potentially say its easier to digest if the query is in one place (if each part was doing lots of calculations I might have thought differently though):

```
def build_dsl_search_query(query, search_fields, sorting_orders, transferring_body_id=None):
    filter_clauses = []
    aggregation_query =   {"aggs":
         "aggregate_by_transferring_body": {
                "terms": {"field": "transferring_body_id.keyword"},
                "aggs": {
                    "top_transferring_body_hits": {
                        "top_hits": {
                            "size": 1,
                            "_source": ["transferring_body"],
                        }
                    }
                },
            }
        },
    }
    if transferring_body_id:
        filter_clauses = (
            [{"term": {"transferring_body_id.keyword": _id}}]
        )
        aggregation_query = None

    dsl_query = {
        "query": {
            "bool": {
                "must": [
                     {
                        "multi_match": {
                             "query": query,
                            "fields": fields,
                             "operator": "AND",
                             "fuzziness": "AUTO",
                             "lenient": True,
                        }
                     }
                 ],
                "filter": filter_clauses,
            }
        },
        "sort": sorting_query,
        "_source": {"exclude": ["*.keyword"]},
        "highlight": {
            "pre_tags": ["<mark>"],
            "post_tags": ["</mark>"],
            "fields": {
                "*": {},
            },
        },
    }
    if aggregations:
        dsl_query.extend(aggregations)

    return dsl_query
```

in routes:
```
search_fields = get_open_search_fields_to_search_on()
sorting_orders = build_sorting_orders_open_search(request.args)
search_query = build_dsl_search_query(query, search_fields, sorting_orders, transferring_body_id) # (or just search_query = build_dsl_search_query(query, search_fields, sorting_orders) for the summary results one)
````

then all three of these functions can be unit tested , and now `build_dsl_search_query` is really simple where it only has the logic at the top based off passing in transferring_body_id or not.

you could even take it a step further and have 

```
def build_dsl_search_query(query, search_fields, sorting_orders, filter_clauses)
    return {
        "query": {
            "bool": {
                "must": [
                     {
                        "multi_match": {
                             "query": query,
                            "fields": fields,
                             "operator": "AND",
                             "fuzziness": "AUTO",
                             "lenient": True,
                        }
                     }
                 ],
                "filter": filter_clauses,
            }
        },
        "sort": sorting_query,
        "_source": {"exclude": ["*.keyword"]},
        "highlight": {
            "pre_tags": ["<mark>"],
            "post_tags": ["</mark>"],
            "fields": {
                "*": {},
            },
        },
    }
```

```
def build_search_results_summary_query(query, search_fields, sorting_orders):
    filter_clauses = []
    dsl_query = build_dsl_search_query(query, search_fields, sorting_orders, filter_clauses)
    aggregation_query =   {"aggs":
         "aggregate_by_transferring_body": {
                "terms": {"field": "transferring_body_id.keyword"},
                "aggs": {
                    "top_transferring_body_hits": {
                        "top_hits": {
                            "size": 1,
                            "_source": ["transferring_body"],
                        }
                    }
                },
            }
        },
    }

    dsl_query.extend(aggregations)

    return dsl_query
````


```
def build_search_transferring_body_query(query, search_fields, sorting_orders, transferring_body_id):
    filter_clauses = [{"term": {"transferring_body_id.keyword": _id}}]
    dsl_query = build_dsl_search_query(query, search_fields, sorting_orders, filter_clauses)
    return dsl_query
````